fetch-array-buffer/index.html

```diff
@@ -23,59 +20,56 @@ <h1>Fetch arrayBuffer example</h1>
     <pre></pre>
   </body>
   <script>
-    // define variables
-    var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
-    var source;
+    let source;
```

Comment:
This looks hacky. Return `source` from the `getData` final promise handler?

Reply:
Done.
I have:
- used the modern constructor `new AudioBufferSourceNode` instead of the deprecated `createBufferNode()` factory method.
- Transformed the "global" `let source` into a `const source` inside the final `getData()` promise handler and returned it.
- Promisifying (nice word!) `decodeAudioData()`.

This means we also modernize the use of the Web Audio API (that was created at a time constructors weren't much used in Web APIs and before the addition of promises to JS). Which is great too.